### DISCUSSION TITLE
Python Script for Sparse Indexes

### DISCUSSION THREAD

**Discussion by hieudx149** [castorini/pyserini#1217] [Q&A] [answered]
Hi there,
I read your document about "How do I index and search my own documents",  in [here](https://github.com/castorini/pyserini#how-do-i-index-and-search-my-own-documents) you guys provide some bash script guide to indexing and batch searching documents, can you guide me to run indexing and batch searching in python like this:
![image](https://user-images.githubusercontent.com/61724310/174989311-b33eab6b-4650-48d5-a65b-b682006aaba7.png)

**Comment by lintool** [comment, marked as answer]
Unfortunately, there's no programmatic way to perform indexing in Python, since the indexer is in Java. The Python command simply wraps a Java invocation.
